x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -6912,12 +7066,23 @@ private EsQueryExec assertChildIsGeoPointExtract(UnaryExec parent, boolean useDo
     }
 
     private EsQueryExec assertChildIsExtractedAsDocValues(UnaryExec parent, boolean useDocValues, DataType dataType) {
+        // TODO(gal) why is this OK To vacuously true?
         var extract = as(parent.child(), FieldExtractExec.class);
+        assertThat(extract.boundAttributes(), is(empty()));
         assertTrue(
             "Expect field attribute to be extracted as " + (useDocValues ? "doc-values" : "source"),
             extract.attributesToExtract()
                 .stream()
-                .allMatch(attr -> extract.hasDocValuesAttribute(attr) == useDocValues && attr.dataType() == dataType)
+                .allMatch(attr -> extract.docValuesAttributes().contains(attr) == useDocValues && attr.dataType() == dataType)
+        );
+        return source(extract.child());
+    }
+
+    private static EsQueryExec assertChildIsExtractedAsBounds(UnaryExec parent, DataType dataType) {
```

Comment:
I presume we could have combined these two methods, and replace the boolean useDocValues with a FieldExtractPreference, which would just switch which collection to test.

Reply:
Pretty large refactoring, but I've replaced all the `Boolean`s with the enum 😎.